pkg/db/build_store.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package db
+
+import (
+	"sync"
+
+	"github.com/daytonaio/daytona/pkg/build"
+	. "github.com/daytonaio/daytona/pkg/db/dto"
+	"gorm.io/gorm"
+)
+
+type BuildStore struct {
+	db   *gorm.DB
+	Lock sync.Mutex
+}
+
+func NewBuildStore(db *gorm.DB) (*BuildStore, error) {
+	err := db.AutoMigrate(&BuildDTO{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &BuildStore{db: db}, nil
+}
+
+func (b *BuildStore) Find(hash string) (*build.Build, error) {
+	b.Lock.Lock()
+	defer b.Lock.Unlock()
+
+	buildDTO := BuildDTO{}
+	tx := b.db.Where("hash = ?", hash).First(&buildDTO)
+	if tx.Error != nil {
+		if tx.Error == gorm.ErrRecordNotFound {
+			return nil, build.ErrBuildNotFound
+		}
+		return nil, tx.Error
+	}
+
+	build := ToBuild(buildDTO)
+
+	return build, nil
+}
+
+func (b *BuildStore) FindAllByState(state build.BuildState) ([]*build.Build, error) {
```

Comment:
I suggest adding a `Filter` struct here instead of one argument. A struct is more extensible. If we were to introduce another filter, we would have to modify all callers of the method if we go the argument route.